ykrt/src/deopt.rs

```diff
@@ -204,14 +204,11 @@ extern "C" fn ts_reconstruct(ctx: *mut c_void, module: LLVMModuleRef) -> LLVMErr
                 let _val = unsafe { registers.get(*reg) };
                 todo!();
             }
-            SMLocation::Direct(reg, off, _size) => {
-                // When using `llvm.experimental.deoptimize` then direct locations should always be
-                // in relation to RBP.
-                assert_eq!(*reg, 6);
-                let addr = unsafe { registers.get(*reg) as *mut u8 };
-                let addr = unsafe { addr.offset(isize::try_from(*off).unwrap()) };
-                let aot = &aotvals[i];
-                framerec.var_init(aot.val, aot.sfidx, addr as u64);
+            SMLocation::Direct(..) => {
+                // Normally a `Direct` location would be used for a local variable expressing a
+                // pointer to the frame. Since JITted code frames have had their stack "unified"
```

Comment:
Maybe we can just directly say that we have a shadow stack that is shared between AOT and JIT, so readers don't need to read between the lines.